libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -121,11 +141,25 @@ bool HTTPClient::begin(String url, String httpsFingerprint)
 bool HTTPClient::begin(String url)
 {
     _transportTraits.reset(nullptr);
-    _port = 80;
-    if (!beginInternal(url, "http")) {
+    if (!beginInternal(url, nullptr)) {
+        return false;
+    }
+    if (_protocol == "https" || _port == 443) {
+        _port = 443;
+        _transportTraits = TransportTraitsPtr(new TLSTraits());
+    }
+    else {
+        _port = 80;
+        _transportTraits = TransportTraitsPtr(new TransportTraits());
+    }
+    _tcp = _transportTraits->create();
+
+    if(!_tcp->connect(_host.c_str(), _port)) {
```

Comment:
This is a fairly big behavioural change, and it's not hard to come up with a bunch of cases when it will break existing sketches. If you need to defer setting the CA until the time connection is established, save the CA information in the appropriate Traits structure (probably a new one, dedicated to verification using a root CA cert). Then use this information when the verification has to take place. If HTTPClient needs to interact with Traits class at some other point in the process, add a new interface method and implement it as needed. It doesn't look like it is worth changing the behaviour of user-facing class, when the problem can be solved inside the implementation.